specifyweb/frontend/js_src/lib/components/FormEditor/__tests__/createView.test.ts

```diff
@@ -489,7 +489,7 @@ test('Add new view definition based on existing', () =>
         name: 'CollectionObjectAttachment Table (2)',
         raw: `<viewdef>
   <desc>ObjectAttachment grid view.</desc>
-  <definition>ObjectAttachment Form</definition>
+  <definition>CollectionObjectAttachment Table (2)</definition>
```

Comment:
hmm, this change doesn't look right
see https://sp7demofish.specifycloud.org/context/view.json?name=CollectionObjectAttachment - it is set to "ObjectAttachment Form" in there at the moment

Reply:
note that the attachment table may be special.

could you check on some attachment, and non-attachment form that has formtable to see what the expected value for `<definition>` is?
then compare what `createViewFromTemplate` would output before this change and after the change and see which is correct

Reply:
okay. I think I know what's happening now.

The way `createViewFromTemplate` worked before wasn't correct - I forgot to make the viewdef name in the `<definition>` unique

And the way it works after you change isn't quite correct either - if you look at the viewdef after your code changes:
```js
        name: 'CollectionObjectAttachment Table (2)',
        raw: `<viewdef>
  <desc>ObjectAttachment grid view.</desc>
  <definition>CollectionObjectAttachment Table (2)</definition>
```
it is saying that when the form finds viewdef named "CollectionObjectAttachment Table (2)" (coming from the name attribute), it should use the "CollectionObjectAttachment Table (2)" viewdef (coming from the definition) - which would create a cycle

instead, it should be taking the `<definition>` value that was there before, and making that unique
thus, you need to change `definition.name` in:
https://github.com/specify/specify7/pull/4144/files#diff-95191ed3f14da84d43cc55789dfd7fb11c121532e0fc5615b08b1ede1e58f61cR312-R314
to `subChild.string`. i.e so that it looks like this:
```ts
                 string: nameMapper[subChild.string as LocalizedString] ??  subChild.string,
```